File: java/Implement Stack using Queues.java
//push: O(1)，pop: O(n)，top: O(n)

private Queue<Integer> q = new LinkedList<Integer>();
    
    // Push element x onto stack.
    public void push(int x) {
        q.offer(x);
    }

    // Removes the element on top of the stack.
    public void pop() {
        int size = q.size();
        for(int i = 1; i<size;i++){
            q.offer(q.poll());
        }
        q.poll();
    }

    // Get the top element.
    public int top() {
        int size = q.size();
        for(int i = 1; i<size;i++){
            q.offer(q.poll());
        }
        int value = q.peek();
        q.offer(q.poll());
        
        return value;
    }

    // Return whether the stack is empty.
    public boolean empty() {
        return q.isEmpty();
    }

File: C_plus/Implement Stack using Queues.cpp
public:

    queue<int> q1 ;
    queue<int> q2 ;
    

    // Push element x onto stack.
    void push(int x) {
        
        q2.push(x);
        
        while(!q1.empty()) {
            
            int value = q1.front();
            
            q1.pop();
            
            q2.push(value);
            
        }
        
        queue<int> tmp = q1;
        q1 = q2;
        q2 = tmp;
         
    }

    // Removes the element on top of the stack.
    void pop() {
        
        q1.pop();
        
    }

    // Get the top element.
    int top() {
        
        return q1.front();
        
    }

    // Return whether the stack is empty.
    bool empty() {
        
        return q1.empty();
        
    }
